### ISSUE TITLE
DropDown; Lookup; How to define which field (value) from the "Lookup row" is effectively stored into the field to which the lookup is attached

### ISSUE THREAD

**Issue by JohnRanger** [volkanceylan/Serenity#2453]
Dear all,

**Challenge:**
I  have the situation that I want to store in the form field of a dialog the value of a field (from a lookup) that is different than the identity field. By default, the lookup always stores the identity field value of the selected item from the dropdown (but returns actually all fields which are defined in XYXZrow.cs with [LookupInclude]).

The lookupEditor looks like this:

```ts
   // *** Print-Queues for WorkstationGroup ***
    @Serenity.Decorators.registerEditor()
    export class PrintQueuesforWorkstationGroupEditor extends
        Serenity.LookupEditorBase<Serenity.LookupEditorOptions, SetDefaultPrinter.LinkedQueuesRow> {

        constructor(container: JQuery, opt: Serenity.LookupEditorOptions) {
            super(container, opt);
        }

        protected getLookupKey() {
            return SetDefaultPrinter.LinkedQueuesRow.lookupKey;
        }

        protected getItemText(item: SetDefaultPrinter.LinkedQueuesRow, lookup: Q.Lookup<SetDefaultPrinter.LinkedQueuesRow>) {
            return item.PrintQueueComment + ' --> ( ' + item.PrintQueueName + ' )';
        }

        protected getItems(lookup: Q.Lookup<SetDefaultPrinter.LinkedQueuesRow>) {

            var output = super.getItems(lookup).sort(function (a, b) {return a.PrintQueueComment.localeCompare(b.PrintQueueComment);});
            return output;
        }

    }
```

I didn't find any example for this and also didn't find a property within the lookupOptions to define the to-be-returned field from which the value is retrieved and to be stored in the form field.

**Question:**
How can I "tell" the form field (or the lookupEditor) which **field**'s value is be stored within the form field?

P.S: I know that I can workarround this with using an In-Memory form field (where I attach the lookup) and exrtract the desired field's value with a search via the Identity value and then write this in the beforeSave event into the DB but there should (must :-) ) be a bether way...

Thanks for your help and

with kind regards,

John

**Comment by volkanceylan** [comment]
You can't use another field as lookup editors "value". Lookup editor has to search the record later by this "value", what if there are multiple items that has that "value".

I'd recommend creating a new custom Lookup, similar to one in Northwind / Country list.

**Comment by JohnRanger** [comment]
Good evening Volkan,

thanks for your valuable Explanation. In the end I did a complete redesign of the Tables in question and pulled them together on the SQL side via views.

Thanks again and

with Kind regards,

John

**Comment by stixoffire** [comment]
@JohnRanger  - I did this by changing the name field as I did not want the integer value I wanted the text value .. I had a table of US States and also Countries where I did this ..  it works but was a compromise.

I am curious though because you mentioned views - and I am searching for how I can have a drop down filter on my view column. 
I have an orders receiving View  has Item, Quantity, Vendor etc... that I want to filter on vendor via a Lookup drop down. I placed LookupInclude(typeof(VendorRow)) but I have nothing populated in the drop down.
